modules/posix/src/ciaaI2CDevices.c

```diff
@@ -234,271 +245,128 @@ extern int32_t ciaaI2CDevices_ioctl(ciaaDevices_deviceType const * const device,
-         rawCount = ciaaLibs_circBufRawCount(cbuf, tail);
-
-         /* write more bytes */
-         write = I2CDevice->device->write(device->loLayer,
-               ciaaLibs_circBufReadPos(cbuf), rawCount);
-
-         if (write > 0)
-         {
-            /* update buffer */
-            ciaaLibs_circBufUpdateHead(cbuf, write);
-         }
-      }
-      /* if task is blocked and waiting for reception of this device */
-      if ( (255 != taskID) &&
-            (I2CDevice->blocked.fct ==
-             (void*) ciaaI2CDevices_write) )
-      {
-         /* invalidate task id */
-         I2CDevice->blocked.taskID = 255; /* TODO add a macro */
+	   width = I2CDevice->masterData.registerAddWidth;
+	   pByte = (uint8_t *)&I2CDevice->masterData.registerAdd;
+	   for( i = 0; i < width; i++ )
+	   {
+		   val[i] = pByte[width - 1 - i];
+	   }
+	   ciaaLibs_circBufPut(cbuf, val, width);
 
-         /* reset function */
-         I2CDevice->blocked.fct = NULL;
+	   ciaaLibs_circBufPut(cbuf, buf, nbyte);
 
-         /* set task event */
-#ifdef POSIXE
-         SetEvent(taskID, POSIXE);
-#endif
-      }
-   }
-}
+	   ret = I2CDevice->device->write(device->loLayer, ciaaLibs_circBufReadPos(cbuf), (nbyte + I2CDevice->masterData.registerAddWidth));
 
-extern void ciaaI2CDevices_rxIndication(ciaaDevices_deviceType const * const device, uint32_t const nbyte)
-{
-   /* get I2C device */
-   ciaaI2CDevices_deviceType * I2CDevice =
-      (ciaaI2CDevices_deviceType*) device->layer;
-   ciaaLibs_CircBufType * cbuf = &I2CDevice->rxBuf;
-   uint32_t head = cbuf->head;
-   uint32_t rawSpace = ciaaLibs_circBufRawSpace(cbuf, head);
-   uint32_t space = ciaaLibs_circBufSpace(cbuf, head);
-   uint32_t read = 0;
-   TaskType taskID = I2CDevice->blocked.taskID;
-
-   read = I2CDevice->device->read(device->loLayer, ciaaLibs_circBufWritePos(cbuf), rawSpace);
-
-   /* if rawSpace is full but more space is avaialble */
-   if ((read == rawSpace) && (space > rawSpace))
-   {
-      read += I2CDevice->device->read(
-            device->loLayer,
-            &cbuf->buf[0],
-            space - rawSpace);
-   }
-   else
-   {
-      if ((read == rawSpace) && (space <= rawSpace))
-      {
-         /* data may be lost because not place on the receive buffer */
-         /* TODO */
-      }
-      else
-      {
-         /* read less bytes than provided */
-         /* nothing to do */
-      }
+	   ciaaLibs_circBufClean(cbuf);
    }
 
-   /* update tail */
-   ciaaLibs_circBufUpdateTail(cbuf, read);
-
-   /* if data has been read */
-   if ( (0 < read) && (255 != taskID) &&
-         (I2CDevice->blocked.fct == (void*) ciaaI2CDevices_read ) )
-   {
-      /* invalidate task id */
-      I2CDevice->blocked.taskID = 255; /* TODO add macro */
-
-      /* reset blocked function */
-      I2CDevice->blocked.fct = NULL;
-
-#ifdef POSIXE
-      /* set task event */
-      SetEvent(taskID, POSIXE);
-#endif
-   }
+   return ret;
 }
 
 /** @} doxygen end group definition */
```

Comment:
por que no agregas ahora la linea? es hacer copy and paste de  /** @} doxygen end group definition */